tendrl/gluster_integration/sds_sync/utilization.py

```diff
@@ -40,27 +40,52 @@ def sync_utilization_details(volumes):
     out = ''
     try:
         out, err = cmd.communicate()
+
         util_det = json.loads(out)
-        for k, v in util_det.iteritems():
-            volume = voldict[k]
-            volume.usable_capacity = int(v['total'])
-            volume.used_capacity = int(v['used'])
-            volume.pcnt_used = str(v['pcnt_used'])
-            volume.total_inode_capacity = int(v['total_inode'])
-            volume.used_inode_capacity = int(v['used_inode'])
-            volume.pcnt_inode_used = str(v['pcnt_inode_used'])
-            volume.save()
-            cluster_used_capacity += volume.used_capacity
-            cluster_usable_capacity += volume.usable_capacity
-    except(KeyError, TypeError, ValueError, AttributeError):
+        if err != '':
+            logger.log(
+                "error",
+                NS.publisher_id,
+                {'message' : 'Error : %s\n' %err}
+            )
+        if bool(util_det):
+            if util_det['error']:
+                logger.log(
+                    "error",
+                    NS.publisher_id,
+                    {'message' : 'Error getting utilization of some volumes\nError: %s\n' % util_det['error']}
+                )
+
+            if util_det['success']:
+                for item in util_det['success']:
+                    for k, v in item.iteritems():
+                        try:
+                            volume = voldict[k]
+                            volume.usable_capacity = int(v['total'])
+                            volume.used_capacity = int(v['used'])
+                            volume.pcnt_used = str(v['pcnt_used'])
+                            volume.total_inode_capacity = int(v['total_inode'])
+                            volume.used_inode_capacity = int(v['used_inode'])
+                            volume.pcnt_inode_used = str(v['pcnt_inode_used'])
+                            volume.save()
+                            cluster_used_capacity += volume.used_capacity
+                            cluster_usable_capacity += volume.usable_capacity
+
+                        except(KeyError, TypeError, ValueError, AttributeError) as e:
+                            logger.log(
+                                "error",
+                                NS.publisher_id,
+                                {'message' : "Error parsing utilization data of volume '%s'\nResult: %s\nError: %s\n" % (k, v, e)}
+                            )
+    except:
         logger.log(
             "error",
             NS.publisher_id,
             {
-                "message": "Error getting utilization of "
-                "volumes %s . result: %s . err: %s" % (volnames, out, err)
+                "message": "Error getting volume utilization.\nResult: %s\n" % out
```

Comment:
`out` might not be populated in some cases of exception I feel, so is it good to add this to log here?

Reply:
I think there are only 2 cases(non ideal) to reach this exception.
1. when vol_utilization fails to execute and 'out' is empty - in this case I feel rather than just telling 'Error getting volume utilization', it is cleaner to say that we got an empty result which suggests that vol_utilization is failing.
2. when vol_utilization dumps invalid json to stdout - here 'out' would be populated but can not be loaded as json to util_det. So, I think it is clearer to show that we are getting invalid json which is causing this exception.

overall, it is about perception I believe. Only time this might add value is during tracing the logs.

Reply:
I would agree with shubhendu here, if out can have multiple values or empty string, then checking those things with if conditions is better and then log your messages accordingly. A message with " Result : " wont look good. Being explicit with exact error and reason would be better.

Reply:
I will just remove 'Out' and keep the log generic?
another problem is, Codacy expects me to not have a global `except` which is not possible here as we don't know what specific exceptions would be thrown?